src/app/Dashboard/Diagnostics/DiagnosticsCard.tsx

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright The Cryostat Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  DashboardCardTypeProps,
+  DashboardCardFC,
+  DashboardCardSizes,
+  DashboardCardDescriptor,
+} from '@app/Dashboard/types';
+import { ErrorView } from '@app/ErrorView/ErrorView';
+import { FeatureLevel } from '@app/Shared/Services/service.types';
+import { LoadingProps } from '@app/Shared/Components/types';
+import { ServiceContext } from '@app/Shared/Services/Services';
+import { useSubscriptions } from '@app/utils/hooks/useSubscriptions';
+import {
+  Bullseye,
+  Button,
+  CardBody,
+  CardHeader,
+  CardTitle,
+  EmptyState,
+  EmptyStateBody,
+  EmptyStateIcon,
+  EmptyStateVariant,
+  Label,
+  EmptyStateHeader,
+  EmptyStateFooter,
+} from '@patternfly/react-core';
+import { DataSourceIcon, SyncAltIcon, TachometerAltIcon } from '@patternfly/react-icons';
+import * as React from 'react';
+import { Trans, useTranslation } from 'react-i18next';
+import { DashboardCard } from '../DashboardCard';
+
+export interface DiagnosticsCardProps extends DashboardCardTypeProps {
+  chartKind: string;
+  duration: number;
+  period: number;
+}
+
+export enum DiagnosticsCardKind {}
+
+export function kindToId(kind: string): number {
+  return DiagnosticsCardKind[kind];
+}
+
+export const DiagnosticsCard: DashboardCardFC<DiagnosticsCardProps> = (props) => {
+  const { t } = useTranslation();
+  const serviceContext = React.useContext(ServiceContext);
+  const addSubscription = useSubscriptions();
+  const [errorMessage, setErrorMessage] = React.useState('');
+  const [running, setRunning] = React.useState(false);
```

Comment:
eslint reveals `setRunning` is never used - I think it should be called `setRunning(true)` immediately at the top of `handleGC`, and `setRunning(false)` within the service subscription there (and within the error handler).